aws/session/internal/ini/visitor.go

```diff
@@ -0,0 +1,135 @@
+package ini
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+)
+
+// Visitor is an interface used by walkers that will
+// traverse an array of ASTs.
+type Visitor interface {
+	VisitExpr(AST) error
+	VisitStatement(AST) error
+}
+
+// Sections is a map of Section structures that represent
+// a configuration.
+type Sections map[string]Section
```

Comment:
with the accessors being preferred way of getting the section, moving the `map[string]section` to a unexpected field would be good.